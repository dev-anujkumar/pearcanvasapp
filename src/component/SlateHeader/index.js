import SlateHeader from './SlateHeader.jsx';
export default SlateHeader;