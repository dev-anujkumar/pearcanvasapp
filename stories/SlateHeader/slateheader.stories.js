import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import  SlateHeader  from '../../src/component/SlateHeader/slateHeader';

storiesOf('SlateHeader', module)
.add('Slate Header', () => <SlateHeader onClick={action('clicked')} />);