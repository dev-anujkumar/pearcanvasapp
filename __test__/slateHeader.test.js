import React from 'react';
import SlateHeader from '../src/component/SlateHeader/slateHeader.jsx';


describe('Testing SlateHeader component with props', () => {

    it('render SlateHeader component ', () => {
        const component = mount(<SlateHeader />);
        expect(component).toMatchSnapshot();
    })

    it('render container-introduction Slate ', () => {
        const component = mount(<SlateHeader slateType='container-introduction' />);
        expect(component).toMatchSnapshot();
    })

    it('render assessment Slate ', () => {
        const component = mount(<SlateHeader slateType='assessment' />);
        expect(component).toMatchSnapshot();
    })

    it('render section Slate ', () => {
        const component = mount(<SlateHeader slateType='section' />);
        expect(component).toMatchSnapshot();
    })

    it('onClick Event', () => {
        const component = mount(<SlateHeader />);
        component.find('div#backward-nav-active').simulate('click');
        component.find('div#forward-nav-active').simulate('click');
    })
})