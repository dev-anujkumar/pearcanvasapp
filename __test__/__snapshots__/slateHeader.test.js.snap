// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SlateHeader component with props render SlateHeader component  1`] = `
<SlateHeader>
  <div
    className="slate-Title"
  >
    <div
      className="canvas-header"
      id="canvas-header"
    >
      <label
        className="header-label"
      >
        SLATE:
      </label>
      <div
        className="input-text"
      >
        <label
          className="u-hidden"
          htmlFor="txt-input"
        />
        <input
          className="txt-input"
          disabled={true}
          placeholder="title"
          type="text"
        />
      </div>
      <Button
        labelText="P"
        onClick={[Function]}
        type="backward-nav-active"
      >
        <span
          className="btn-element"
          id="backward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </Button>
      <Button
        labelText="P"
        onClick={[Function]}
        type="forward-nav-active"
      >
        <span
          className="btn-element"
          id="forward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </Button>
    </div>
  </div>
</SlateHeader>
`;

exports[`Testing SlateHeader component with props render assessment Slate  1`] = `
<SlateHeader
  slateType="assessment"
>
  <div
    className="slate-Title"
  >
    <div
      className="canvas-header"
      id="canvas-header"
    >
      <label
        className="header-label"
      >
        ASSESSMENT SLATE:
      </label>
      <div
        className="input-text"
      >
        <label
          className="u-hidden"
          htmlFor="txt-input"
        />
        <input
          className="txt-input"
          disabled={true}
          placeholder="title"
          type="text"
        />
      </div>
      <Button
        labelText="P"
        onClick={[Function]}
        type="backward-nav-active"
      >
        <span
          className="btn-element"
          id="backward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </Button>
      <Button
        labelText="P"
        onClick={[Function]}
        type="forward-nav-active"
      >
        <span
          className="btn-element"
          id="forward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </Button>
    </div>
  </div>
</SlateHeader>
`;

exports[`Testing SlateHeader component with props render container-introduction Slate  1`] = `
<SlateHeader
  slateType="container-introduction"
>
  <div
    className="slate-Title"
  >
    <div
      className="canvas-header"
      id="canvas-header"
    >
      <label
        className="header-label"
      >
        INTRODUCTORY SLATE:
      </label>
      <div
        className="input-text"
      >
        <label
          className="u-hidden"
          htmlFor="txt-input"
        />
        <input
          className="txt-input"
          disabled={true}
          placeholder="title"
          type="text"
        />
      </div>
      <Button
        labelText="P"
        onClick={[Function]}
        type="backward-nav-active"
      >
        <span
          className="btn-element"
          id="backward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </Button>
      <Button
        labelText="P"
        onClick={[Function]}
        type="forward-nav-active"
      >
        <span
          className="btn-element"
          id="forward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </Button>
    </div>
  </div>
</SlateHeader>
`;

exports[`Testing SlateHeader component with props render section Slate  1`] = `
<SlateHeader
  slateType="section"
>
  <div
    className="slate-Title"
  >
    <div
      className="canvas-header"
      id="canvas-header"
    >
      <label
        className="header-label"
      >
        SLATE:
      </label>
      <div
        className="input-text"
      >
        <label
          className="u-hidden"
          htmlFor="txt-input"
        />
        <input
          className="txt-input"
          disabled={true}
          placeholder="title"
          type="text"
        />
      </div>
      <Button
        labelText="P"
        onClick={[Function]}
        type="backward-nav-active"
      >
        <span
          className="btn-element"
          id="backward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </Button>
      <Button
        labelText="P"
        onClick={[Function]}
        type="forward-nav-active"
      >
        <span
          className="btn-element"
          id="forward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </Button>
    </div>
  </div>
</SlateHeader>
`;
